File: python/ProgramStudi.py
from Course import Course

class ProgramStudi:
# kelas yang digunakan untuk mengimplentasikan kelas ProgramStudi

    ## inisialiasasi attribute
    __nama_prodi =""
    __kode =""
    __course=Course("")

    # konstruktor
    def __init__(self, nama_prodi, kode, course):
        self.__nama_prodi = nama_prodi
        self.__kode = kode
        self.__course = course

    # setter & setter methods #

    def setNamaProdi(self, nama_prodi):
    # mengset nilai atribut nama
        self.__nama_prodi = nama_prodi

    def setKode(self, kode):
    # mengset nilai atribut kode
        self.__kode = kode

    def setCourse(self, course):
    # mengset nilai atribut kode
        self.__course = course

    def getNamaProdi(self):
    # mengembalikan nilai atribut nama_prodi
        return self.__nama_prodi

    def getKode(self):
    # mengembalikan nilai atribut kode
        return self.__kode
    
    def getCourse(self):
    # mengembalikan nilai atribut course
        return self.__course

File: c++/ProgramStudi.cpp
#include "Course.cpp"

using namespace std;

class ProgramStudi {
    private:
        string nama_prodi;
        string kode;
        list<Course> courseList;
    public:
        ProgramStudi(){};

        ProgramStudi(string namaProdi, string kode, list<Course> courseList) {
            this->nama_prodi = namaProdi;
            this->kode = kode;
            this->courseList = courseList;
        }


        /* Setter */
        void setNamaProdi(string namaProdi) {
            this->nama_prodi = namaProdi;
        }

        void setKode(string kode) {
            this->kode = kode;
        }

        void setCourse(list<Course> courseList) {
            this->courseList = courseList;
        }

        /* Getter */

        string getNamaProdi() {
            return nama_prodi;
        }

        string getKode() {
            return kode;
        }

        list<Course> getCourse() {
            return courseList;
        }

        /* Method print courseList*/
        void printCourseList() {
            for (auto it = courseList.begin(); it != courseList.end(); ++it) {
                cout << " - " << it->getNamaMatkul() << endl;
            }
        }

        /* Destruct*/
        ~ProgramStudi(){}
};
